File: app/src/main/java/com/example/taxconnect/adapter/RatingAdapter.java
package com.example.taxconnect.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.taxconnect.R;
import com.example.taxconnect.model.RatingModel;

import java.util.ArrayList;
import java.util.List;

public class RatingAdapter extends RecyclerView.Adapter<RatingAdapter.RatingViewHolder> {

    private List<RatingModel> ratings = new ArrayList<>();

    public void setRatings(List<RatingModel> ratings) {
        this.ratings = ratings;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public RatingViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_rating, parent, false);
        return new RatingViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RatingViewHolder holder, int position) {
        RatingModel rating = ratings.get(position);
        holder.bind(rating);
    }

    @Override
    public int getItemCount() {
        return ratings.size();
    }

    static class RatingViewHolder extends RecyclerView.ViewHolder {
        TextView tvUserName, tvRatingValue, tvReview;

        public RatingViewHolder(@NonNull View itemView) {
            super(itemView);
            tvUserName = itemView.findViewById(R.id.tvUserName);
            tvRatingValue = itemView.findViewById(R.id.tvRatingValue);
            tvReview = itemView.findViewById(R.id.tvReview);
        }

        public void bind(RatingModel rating) {
            tvUserName.setText(rating.getUserName() != null ? rating.getUserName() : "Anonymous");
            tvRatingValue.setText(String.valueOf(rating.getRating()));
            tvReview.setText(rating.getReview() != null ? rating.getReview() : "");
        }
    }
}

File: tax/app/src/main/java/com/example/taxconnect/features/ca/RatingAdapter.kt
package com.example.taxconnect.features.ca

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.taxconnect.R
import com.example.taxconnect.data.models.RatingModel
import java.util.*

class RatingAdapter : RecyclerView.Adapter<RatingAdapter.RatingViewHolder>() {

    private var ratings: List<RatingModel> = ArrayList()

    fun setRatings(ratings: List<RatingModel>) {
        this.ratings = ratings
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RatingViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_rating, parent, false)
        return RatingViewHolder(view)
    }

    override fun onBindViewHolder(holder: RatingViewHolder, position: Int) {
        holder.bind(ratings[position])
    }

    override fun getItemCount(): Int = ratings.size

    class RatingViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvUserName: TextView = itemView.findViewById(R.id.tvUserName)
        private val tvRatingValue: TextView = itemView.findViewById(R.id.tvRatingValue)
        private val tvReview: TextView = itemView.findViewById(R.id.tvReview)

        fun bind(rating: RatingModel) {
            tvUserName.text = rating.userName ?: "Anonymous"
            tvRatingValue.text = String.format(Locale.getDefault(), "%.1f", rating.rating)
            tvReview.text = rating.review ?: ""
            tvReview.visibility = if (rating.review.isNullOrEmpty()) View.GONE else View.VISIBLE
        }
    }
}
